### PR TITLE
HTTParty::RedirectionTooDeepIntegrated::BaseAppWorker@default: HTTP redirects too deep

### PR DESCRIPTION
**Pull request by emaglio**
- Existing tests fixed
- Test Added for Tag
- fixed issue with `HTTParty::RedirectionTooDeep`

### CODE REVIEW

File: lib/convertkit/client.rb [hunk 1]
```diff
@@ -24,12 +24,16 @@ def get_request(path)
...
-    def post_request(path, params)
+    def post_request(path, params, follow_redirects: false)
       url = "#{@uri}/v#{@version}#{path}?from=clickfunnels"
 
-      json = self.class.post(url, body: {
-        api_key: @key,
-        }.merge(params))
+      json = self.class.post(
+        url,
+        follow_redirects: follow_redirects,
```
Comment (emaglio) [inline lib/convertkit/client.rb:32]: this is one of the option that `httparty` has as default `true` so it tries to redirect 5 times and then returns the error. In our case because the `convertkit` post request does not redirect anywhere we just set it to false. Hope it makes sense

Here the links for `httparty`:
- default options: https://github.com/jnunemaker/httparty/blob/v0.16.2/lib/httparty/request.rb#L58
- `handle_response`: https://github.com/jnunemaker/httparty/blob/v0.16.2/lib/httparty/request.rb#L336
- `response_redirect?`: https://github.com/jnunemaker/httparty/blob/v0.16.2/lib/httparty/request.rb#L381

File: spec/convertkit/client_spec.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@ module ConvertKit
       end
 
       it "uses version 2" do
-        expect(client.version).to be(2)
+        expect(client.version).to be(3)
```
Comment (jasich) [inline spec/convertkit/client_spec.rb:17]: Maybe the `it` block title should be updated to `3`?

File: spec/convertkit/form_spec.rb [hunk 3]
```diff
@@ -3,6 +3,53 @@
...
+    let(:form_list) do
+      {
+        "forms": [
+          {
+            "id": 4812649,
+            "subscriber_count": 0,
+            "created_at": "2014-12-20T17:49:34Z",
+            "updated_at": "2014-12-20T17:49:44Z",
+            "name": "Test form",
+            "details": "https://api.convertkit.com/forms/4812649?k=f9922361d11e7339cc1cde3d54c071&v=2",
```
Comment (jasich) [inline spec/convertkit/form_spec.rb]: Not sure if it's related, but is the `v` query param the version.  If so should it be updated to `3` also?

**Review by jasich — CHANGES_REQUESTED** [review]
I think this makes sense, the only thing I'm wondering about is the change from `name` to `first_name`. Is that required? 

Otherwise I think this just needs the `it` block title updated.

File: lib/convertkit/form.rb [hunk 4]
```diff
@@ -43,7 +44,7 @@ class Client
     def forms()
       raw   = get_request("/forms")
       forms = []
-      return forms unless raw["forms"].present?
+      return forms unless raw["forms"].any?
```
Comment (sergeychernyakov) [inline lib/convertkit/form.rb]: what is the sense of this and below change?

File: lib/convertkit/form.rb [hunk 5]
```diff
@@ -27,14 +27,15 @@ def load(data, client)
...
-    
+
     def initialize(id, client)
       @id     = id
       @client = client
     end
 
     def subscribe(params)
-      opts = { email: nil, name: nil, course_opted: true }.merge(params)
+      opts = { email: nil, first_name: nil, course_opted: true }.merge(params)
```
Comment (sergeychernyakov) [inline lib/convertkit/form.rb]: why did you change it on first_name? where is the last_name then?

File: lib/convertkit/form.rb (hunk 4 above)
Comment (sergeychernyakov) [reply lib/convertkit/form.rb]: nil.any? => NoMethodError: undefined method `any?' for nil:NilClass
nil.present? => false

**Review by sergeychernyakov — COMMENTED** [review]
looks good with comments

File: lib/convertkit/form.rb (hunk 4 above)
Comment (emaglio) [reply lib/convertkit/form.rb]: well when you request a list of tags/forms the key `tags/forms` returns always an array, the worse case returns an empty array, but yeah probably is safer to use `present?`

File: spec/convertkit/form_spec.rb (hunk 3 above)
Comment (emaglio) [reply spec/convertkit/form_spec.rb]: The request is stubbed and we set the response so these values are pure random one but I'll fix it for consistency

File: lib/convertkit/form.rb (hunk 4 above)
Comment (emaglio) [reply lib/convertkit/form.rb]: Actually I was playing with it and `.present?` is a rails method not a ruby one....so we can't use it here (I mean we could but I need to add `active_support`) solution is `.nil?`:
```
[1].nil? -> false
[].nil? -> false but then [].each will just return []
nil.nil? -> true
```

**Review by jasich — APPROVED** [review]
